### PR TITLE
Separates saving, preview, publish, and unpublish, plus pretties previews

### LINKED ISSUE #938: CMS: Separating preview, save and publish functions
Generated from #826

For any user, and in any content type, these CTAs should be separate and distinct actions with distinct outcomes

**Preview** : Generates a preview of the content as it will display to site users when published
- For dynamic content types, the preview displays only that content, outside of the context of a page 
- Users should be able to clearly identify how to preview the content, be able to tell the difference between the preview and the editing screen, and be able to move easily between the two views (edit and preview) 
- OOS for now - pages: the preview displays the full page of content (ideally including dynamic content if possible) 

**Save**:  Saves content that has been created or modified, but does **_not_** publish it
- A user should be able to create or modify content, see their changes in the editing screen, save those changes, and return to the item in edit mode to see those changes preserved. 
- If a user modifies content but does not press the Save button, when they go to leave the page, the system should alert them to unsaved changes and give them the option to save. 
- OOS for now - autosave: autosave adds a layer of complexity around state management. It's been desired by forecasters, however with a new system it might not be as bad of a problem. We'll observe with nashville and can include autosave if needed.

**Publish**:  Pushes content to the public-facing site 
- A user must actively select this action through a separate button. It should not be the default state, or achieved through a text box. 
- The publish action should be available only from the edit mode for that content so that users are seeing the content that will be pushed live. 
- A user should be able to publish a content item within the CMS, navigate to the front-facing site, and immediately see the changes live. 

**Unpublish**: Removes content from the public-facing site 
- This option should display on the edit mode for any content that has been published 
- It should not replace the publish button, but exist as a separate action so that a given piece of content can be either unpublished OR modified and published. 
- A user should be able to unpublished a content item within the CMS, navigate to the front-facing site, and immediately see that the content has been removed.

### PR DESCRIPTION
**Pull request by greg-does-weather**
### Separate saving, previewing, publishing, and unpublishing
- Creates a custom template for editing (and creating) content nodes
  - Simplifies it to only the form itself and the action buttons
  - As a result, removes revision history, authorship, and publication checkbox
- Adds a `form_alter` hook to:
  - If a content node is new, set it to unpublished by default
  - If content is being edited and it is currently unpublished, add a publish button to the form
  - If content is being edited and it is currently published, add an unpublish button to the form
  - Ensure that publish and unpublish buttons are always to the left of the delete button
- Adds publish and unpublish functions that are triggered by the buttons above

### Preview
- Creates custom HTML and page templates for preview content. These are essentially copied from our location page templates, except:
  - adds the preview node's "back to editing" link
  - adds a `main--preview` class to the main element, so we can target it with CSS (for reasons that will become clear)
- Updates the `time` field template so all timestamps rendered by Drupal will have our `data-wx-local-time` attribute so they can be targeted by our localization script
- Creates a weather story node template, borrowed directly from the weather story block template
  - **Future thought:** It might be possible to have the weather story block return the node itself, in which case we could render the node directly (e.g., `{{ weather_story }}` instead of `{{ drupal_block("weather_story_block") }}`). If that works, it'll also mean we use the same node template for both use cases, helping ensure they stay in sync. (Yay!)
- Adds a CSS rule to hide node page titles in preview (Told ya.)

### Associated bug fixes
- Updates the location page template to remove `{{ page.content }}` and instead pull in the location search block manually. The only thing we were using `{{ page.content }}` for was to get the location search block, but that's silly and is more fragile.
- Updates the timestamp localization script to work with either Unix epochs or ISO8601 strings
- Updates the entity service that loads weather stories so that it only loads published stories because duh
- Updates the location search block layout so that it only appears on the front page. It is included on location pages by the first item in this bug fix list.

Closes #938 

# What does the reviewer need to know?

@kmranjo I've deployed this (along with your configuration updates and Logan's taxonomy term export) to my environment. I'll send you credentials there so you can see if things are working like you expected. I'll also grab us some time next week to pair up on it for a little bit. This turned out not to be as hard as I expected it to be, so I feel pretty okay with any changes that I need to make from here!

### CODE REVIEW

**Comment by kmranjo** [comment]
@greg-does-weather  This looks good to me, but one quick question. We do have a separate ticket later to tackle the revision history, so we will at some point have to put that back on the page. Knowing that, would it be easier not to remove it now?

**Comment by greg-does-weather** [comment]
Either way should be fine. It's one line of code in the Twig template to put it back. :) If we want to customize how the revision history is displayed to users, it might actually be easiest to completely remove it and then put our own thing in, but I'm not sure.

**Review by eric-gade — APPROVED** [review]
LGTM 👍 
  

## Code Review Checklist

This is an automated comment on every pull request requiring a review. A checked item is either an assertion that I have tested this item or an indication that I have verified it does not apply to this pull request.

## The Basics
- [x] Checks are passing
- [x] I read the code
- [x] I ran the code
- [x] (if applicable) Post deploy steps are run
- [x] (if applicable) I validated the change on the deployed version in <environment>

## Documentation
- [x] changes to “how we do things” are documented in READMEs
- [x] all new functions and methods are commented using plain language
- [x] any new modules added documented in modules.md

## Security
- [x] security false positives are documented
- [x] data from external sources is cleaned and clearly marked

## Reliability
- [x] error handling exists for unusual or missing values
- [x] interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests
- [\x] dependency updates in composer.json also got changed in composer-lock.json

## Infrastructure
- [x] all changes are auditable and documented via a script
- [x] it is clear who can and should run the script
- [x] (if applicable) diagrams have been updated or added in PlantUML

## Accessibility
- [x] New pages have been added to cypress-axe file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [ ] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [ ] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [ ] All form inputs have explicit labels
    - [ ] Form instructions are associated with inputs
    - [ ] All relevant images use an img tag
    - [ ] All images have appropriate alt attributes
    - [ ] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [ ] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [ ] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [ ] Tab order is logical
    - [ ] Tables are coded properly. Tables have proper headers and column attributes
    - [ ] Headings are nested properly. Heading elements are nested in a logical way
    - [ ] Language is set. The language for the page is set
    - [ ] CSS is not required to use the page. The page makes sense with or without CSS
    - [ ] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [ ] Page titles are descriptive

## Device Matrix
- [x] firefox/gecko (renders correctly and user interactions work)
- [x] chrome/chromium/edge (renders correctly and user interactions work)
- [x] safari/webkit (renders correctly and user interactions work)
- [x] web page is readable and usable 
    - [ ] at 480px     (mobile)
    - [ ] at 640px     (tablet)
    - [ ] at 1024px    (desktop)

File: web/themes/new_weather_theme/templates/content/node--weather-story.html.twig [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  set classes = [
+    'node',
+    'node--type-' ~ node.bundle|clean_class,
+    node.isPromoted() ? 'node--promoted',
+    node.isSticky() ? 'node--sticky',
+    not node.isPublished() ? 'node--unpublished',
+    view_mode ? 'node--view-mode-' ~ view_mode|clean_class,
+  ]
+%}
+{{ attach_library('new_weather_theme/node') }}
```
[the remark this answers is not in the document]
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/content/node--weather-story.html.twig:80]: Genuinely unsure. I copied this from the default `node.html.twig` as a template and only changed the parts I needed to.

**Pull request merged**